### PR TITLE
Prevent SSinput from constantly requesting moving in-place

### PR DESCRIPTION
**Pull request by fira**
## About The Pull Request

Just prevents SSinput from constantly calling Move() every tick for every user even when no actual movement is requested. 

Only done rudimentary testing - straightforward and we've been using this for months on CM13 but without diagonal movement. 

I asked the original SSinput author back then which said that it was likely an oversight. I believe moving the diagonal movement direction buffer reset to keyLoop instead of Move() also makes sense as it is actually relevant to SSinput and not client actions as a whole (eg. if you call built-in movement verbs)

## Why It's Good For The Game

Basically performance... This is called every tick for every user, but just imparts an extra proc call and extra checking in client's Move(). It's one of two changes we did on CM13 to lessen the raw performance hit of introducing SSinput, alongside #59339 (which in itself seemed to lessen the calls overhead by about 40%)

## Changelog
:cl:
fix: Input system won't request in-place movements constantly anymore.
/:cl:

### CODE REVIEW

**Comment by BraveMole** [comment]
This is making movement unreliable on TGMC, sometimes you'll randomly skip a tile.

Is this happening on tg or other downstream?

**Comment by optimumtact** [comment]
multiple admins reported the same issue fyi.
